weldx/transformations/local_cs.py

```diff
@@ -683,18 +677,15 @@ def _interp_time_coordinates(self, time: Time) -> Union[xr.DataArray, np.ndarray
             )
             if self.has_reference_time:
                 coordinates.weldx.time_ref = self.reference_time
-        elif self.coordinates.ndim == 1:  # don't interpolate static
-            coordinates = self.coordinates
-        elif time.max() <= self.time.min():  # only use edge timestamp
-            coordinates = self.coordinates.values[0]
-        elif time.min() >= self.time.max():  # only use edge timestamp
-            coordinates = self.coordinates.values[-1]
-        else:  # full interpolation with overlapping times
-            coordinates = ut.xr_interp_coordinates_in_time(self.coordinates, time)
-
-        if len(coordinates) == 1:  # remove "time dimension" for single value cases
```

Comment:
The `len(coordinates) == 1` check can be pretty unstable imo @vhirtham 
at first I was surprised this works without even knowing what it does. Using `len(coordinates.time) == 1` here would probably be more explicit.

Reply:
Well, this is why you find this specific line below the processing of the inputs. `coordinates` and `orientations` should already be formatted correctly to match the specific internal structure I was abusing here. So it is pretty stable I guess unless you change the order of instructions in the `__init__`. But in this case, we have a ton of tests that will fail.

Not sure about `len(coordinates.time) == 1`. You might need to check the existence of `coordinates.time` first since static `coordinates` shouldn't have this attribute. Checking the passed `time` classes length shouldn't help much either since it gives no indication if `coordinates` actually have this extra dimension. I guess we can get it more self-explaining but at the price of additional checks. Not sure if it is worth it for class internal code that already, has a comment to state what it does.

Reply:
> Well, this is why you find this specific line below the processing of the inputs. `coordinates` and `orientations` should already be formatted correctly to match the specific internal structure I was abusing here. So it is pretty stable I guess unless you change the order of instructions in the `__init__`. But in this case, we have a ton of tests that will fail.
> 

I don't think the current `__init__` prevents passing additional dimensions in coordinate dimensions for example?

> Not sure about `len(coordinates.time) == 1`. You might need to check the existence of `coordinates.time` first since static `coordinates` shouldn't have this attribute. Checking the passed `time` classes length shouldn't help much either since it gives no indication if `coordinates` actually have this extra dimension. I guess we can get it more self-explaining but at the price of additional checks. Not sure if it is worth it for class internal code that already, has a comment to state what it does.

wouldn't `len(coordinates) == 1` also fail here if there was no time dimension?
`len(LocalCoordinateSystem().coordinates) == 3`

Reply:
> I don't think the current `__init__` prevents passing additional dimensions in coordinate dimensions for example?

Just checked `_build_coordinates` and it seems that you are correct. Currently, a `DataArray` will be just passed through if I didn't miss something. But I consider this as a bug that should be fixed. The LCS shouldn't store any data that is not necessary to do its job

> wouldn't `len(coordinates) == 1` also fail here if there was no time dimension?
> `len(LocalCoordinateSystem().coordinates) == 3`

No, because the trick here is, that the outer dimension of static coordinates is always of length 3. So if it differs from 3, we have a time dimension and we only want to remove it, if it is of length 1.